### PR TITLE
zone attribute mandatory check removed

### PR DESCRIPTION
**Pull request by venkykuberan**
If this PR is for Terraform, I acknowledge that I have:

- [ ] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [ ] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-downstream-tools), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [X ] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/third_party/terraform/tests) (for handwritten resources or update tests).
- [X ] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/master/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [ ] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/master/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:bug
dataflow - removed required validation for zone for `google_data_flow_job` when region is given in the config 

```
fixes https://github.com/hashicorp/terraform-provider-google/issues/7642

### CODE REVIEW

File: third_party/terraform/resources/resource_dataflow_job.go [hunk 1]
```diff
@@ -515,10 +515,7 @@ func resourceDataflowJobLaunchTemplate(config *Config, project, region, userAgen
 }
 
 func resourceDataflowJobSetupEnv(d *schema.ResourceData, config *Config) (dataflow.RuntimeEnvironment, error) {
-	zone, err := getZone(d, config)
-	if err != nil {
-		return dataflow.RuntimeEnvironment{}, err
-	}
+	zone, _ := getZone(d, config)
```
Comment (ndmckinley) [inline third_party/terraform/resources/resource_dataflow_job.go:518]: Hm, this will work fine if the result coming back is ""?  If so, LGTM.

**Review by ndmckinley — APPROVED** [review]

**Comment by jcanseco** [comment]
I believe this may not work if there is a provider-level zone specified and a region specified for the job, since zone and region cannot be both specified for a job I think.

**Comment by ndmckinley** [comment]
I think I agree that it won't work under those circumstances, but I haven't tested it.  Do you see that as a bug?

**Comment by jcanseco** [comment]
Shouldn't be a bug for us. I was just pointing it out as a potential bug for Terraform users :)

**Pull request merged**